### DISCUSSION TITLE
0.24.4 breaks nested useDismiss behavior when using FloatingFocusManager

### DISCUSSION THREAD

**Discussion by EvHaus** [floating-ui/floating-ui#2425] [Q&A] [answered]
Hey @atomiks,

With the 0.24.4 [change to inert](https://github.com/floating-ui/floating-ui/pull/2399/files), we have a regression in our app where where you can no longer click to dismiss a floating element via `useDismiss()` if that floating element is inside another floating element. It looks like this:
https://github.com/floating-ui/floating-ui/assets/226640/231181f4-750e-48fa-9c3a-276db502f4b3

We have a side flyover element (red), and inside it some floating menu dialogs (yellow). Prior to 0.24.4 when you click outside the floating menu dialog it would get dismissed, but this is no longer happening.

I'm trying to figure out how to make it work with 0.24.4 but I'm struggling. Both of those layers have a `FloatingFocusManager` wrapper around them. Any tips or suggestions?

**Comment by atomiks** [comment]
Does 0.24.5 work? 0.24.4 did have an issue with outsidePress in some scenarios, but maybe there are other ones?

**Reply by EvHaus** [reply]
0.24.5 has the same problem unfortunately

**Reply by atomiks** [reply]
Alright can you convert this into an issue? The problematic code would be [here](https://github.com/floating-ui/floating-ui/blob/03a4f98b86f68c209eeaf67144cfbf5bc28f046c/packages/react/src/hooks/useDismiss.ts#L165-L185) which checks for a marker to determine if the target is inside a third-party element or not (uncontrolled). In your DOM structure, there must be a problem with this check that's not accounted for properly. Can you make a reproduction?

In the meantime, that nested popover can probably use `modal={false}` focus management, which shouldn't have this issue. Or you can stay on 0.24.3 while a fix is worked on.

**Reply by EvHaus** [reply]
Yeah. You got it. I'll make an issue and try to create a repro case for this later today.

**Reply by atomiks** [reply]
No need as I managed to reproduce it. Happens if the top-level floating element is `modal={true}` while the nested one is `modal={false}`, and you click on the top-level popover to dismiss the nested one (rather than outside it)

**Reply by EvHaus** [reply]
Hmm, interesting. In my code, it's the opposite. The outer element is `modal={false}` and the inner one is `modal={true}`.

**Reply by atomiks** [reply]
Interesting; not sure how it can happen in that scenario but either way there's something wrong with the way the attribute gets detected to determine third-party elements

**Reply by atomiks** [reply, marked as answer]
Going to just revert it for now. Not really sure how to do this reliably but there are workarounds for it regardless anyway
